app/Observers/ProductObserver.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace App\Observers;
+
+use App\Models\Product;
+use App\Models\Audit;
+
+class ProductObserver
+{
+    /**
+     * Handle the Product "created" event.
+     */
+    public function created(Product $product): void
+    {
+        Audit::create([
+            // 'user_id'=>1, When seeding product creation does not have a user id - thus set user_id to one when seeding then back to user id thereafter.
+            'user_id'=>auth()->check()?auth()->id() : 1,
```

Comment:
Should we consider not recording audit actions that take place from a seeder? Or perhaps we have a designated 'automated' user in the users table for this purpose?

Maybe simplest option is just to make the user_id nullable - that way, this will not require the ternary.

Can we avoid a majic number?